### DISCUSSION TITLE
How to migrate osu! data to realm?

### DISCUSSION THREAD

**Discussion by Modsnotworking** [ppy/osu#16650] [Q&A] [answered]
How to migrate osu! data to realm? which button should I press to start data migration to realm? this can seem like a dumb question.

**Comment by peppy** [comment, marked as answer]
It's automatic and will run when you update to the new release.

**Reply by Modsnotworking** [reply]
> It's automatic and will run when you update to the new release.

it didn't work after starting the game[Note that I downloaded the new release and installed it]

**Reply by peppy** [reply]
What are you seeing?

**Reply by Modsnotworking** [reply]
> What are you seeing?

I am not seeing the "migration process" on the screen after starting the game. should I download the previous release? will it work if I do that?
[logs.zip](https://github.com/ppy/osu/files/7951466/logs.zip) If the logs help.

**Reply by peppy** [reply]
I'm confused. Your logs show you manually deleting all your beatmaps from the menu. How are you seeing nothing yet doing that?

**Reply by Modsnotworking** [reply]
> Your logs show you manually deleting all your beatmaps from the menu. How are you seeing nothing yet doing that?

yeah I tried to manually migrate my osu! data to realm[I thought that by deleting the beatmaps/everything and then pressing "compact realm" or "block/unblock realm" would start the migration process but it didn't :(] but it looks like it didn't work. So does this mean that I can't migrate osu! to realm by downloading and installing the new/latest release. Also after starting the game I didn't got that Black UI which shows something like "migration in progress" So I didn't got the migration progress UI after starting the game

**Reply by frenzibyte** [reply]
I think you're confused, you either don't have an EF `client.db` database (which can be the case if you were playing around with dev builds prior to release) or the migration was quick you already missed it. If osu! is running at this latest version you're already under Realm, import your data and you're good to go. (also you probaby don't want to touch those memory settings buttons)
